opensquirrel/decomposer/aba_decomposer.py

```diff
@@ -0,0 +1,170 @@
+from __future__ import annotations
+
+import math
+from abc import ABC, abstractmethod
+from collections.abc import Callable
+
+from opensquirrel.common import ATOL
+from opensquirrel.decomposer.general_decomposer import Decomposer
+from opensquirrel.default_gates import Rx, Ry, Rz
+from opensquirrel.ir import Axis, AxisLike, BlochSphereRotation, Float, Gate
+from opensquirrel.utils.identity_filter import filter_out_identities
+
+
+class ABADecomposer(Decomposer, ABC):
+    @property
+    @abstractmethod
+    def ra(self) -> Callable[..., BlochSphereRotation]: ...
+
+    @property
+    @abstractmethod
+    def rb(self) -> Callable[..., BlochSphereRotation]: ...
+
+    _gate_list: list[Callable[..., BlochSphereRotation]] = [Rx, Ry, Rz]
+
+    def __init__(self) -> None:
+        self.index_a = self._gate_list.index(self.ra)
+        self.index_b = self._gate_list.index(self.rb)
+
+    def get_decomposition_angles(self, alpha: float, axis: AxisLike) -> tuple[float, float, float]:
+        """
+        Gives the angles used in the A-B-A decomposition of the Bloch sphere rotation
+        characterized by a rotation around `axis` of angle `alpha`.
+
+        Parameters:
+            alpha: angle of the Bloch sphere rotation
+            axis: _normalized_ axis of the Bloch sphere rotation
+
+        Returns:
+            A triple (theta1, theta2, theta3) corresponding to the decomposition of the
+            arbitrary Bloch sphere rotation into U = Ra(theta3) Rb(theta2) Ra(theta1)
+
+        """
+        axis = Axis(axis)
+
+        if not (-math.pi + ATOL < alpha <= math.pi + ATOL):
+            raise ValueError("Angle needs to be normalized")
+
+        if abs(alpha - math.pi) < ATOL:
+            # alpha == pi, math.tan(alpha / 2) is not defined.
+
+            p: float
+            if abs(axis[self.index_a]) < ATOL:
+                theta2 = math.pi
+                p = 0
+                m = 2 * math.acos(axis[self.index_b])
```

Comment:
```suggestion
            if abs(axis[self.index_a]) < ATOL:
                theta2 = math.pi
                p = 0.0
                m = 2 * math.acos(axis[self.index_b])
```
I think this will work as well.